### PR TITLE
simplify code

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### CODE REVIEW

File: src/raft.rs [hunk 1]
```diff
@@ -751,21 +589,48 @@ impl<T: Storage> Raft<T> {
...
+                    continue 'LOOP;
+                }
             }
-            Some(set) => set
-                .iter()
-                .filter(|id| **id != self_id)
-                .for_each(|id| self.send_append(*id, &mut prs)),
-        };
+            let mut m = Message::default();
+            self.prepare_send_entries(&mut m, prs.get_mut(delegate).unwrap(), vec![]);
```
Comment (Fullstop000) [inline src/raft.rs:641]: Where are the entries sent to the delegate?

File: src/raft.rs [hunk 2]
```diff
@@ -568,6 +568,17 @@ impl<T: Storage> Raft<T> {
...
     }
 
+    fn send_append_with_bcast_targets(&mut self, to: u64, pr: &mut Progress, targets: Vec<u64>) {
+        let old_msg_len = self.msgs.len();
+        self.send_append(to, pr);
+        let new_msg_len = self.msgs.len();
+        if new_msg_len > old_msg_len {
+            self.msgs[old_msg_len].set_bcast_targets(targets);
+        } else {
+            // TODO(qupeng): is it possible?
```
Comment (Fullstop000) [inline src/raft.rs]: I think it's impossible because we just receive the `MsgAppendResponse` from `to`

File: src/raft.rs [hunk 3]
```diff
@@ -1653,10 +1664,14 @@ impl<T: Storage> Raft<T> {
             }
         }
 
-        if send_append && m.from_delegate == INVALID_ID {
-            let from = m.from;
+        // TODO(qupeng): make it correct when `from`'s leader is itself.
+        if send_append && self.groups.get_delegate_by_member(m.from).is_none() {
```
Comment (Fullstop000) [inline src/raft.rs]: Why `self.groups.get_delegate_by_member(m.from).is_none()` ?

Comment (hicqu) [reply src/raft.rs]: If the `append response` comes from follower behind a delegate, we don't need to call `send_append` for it.

Comment (hicqu) [reply src/raft.rs]: The comment is wrong. Should be `from`'s delegate.

Comment (Fullstop000) [reply src/raft.rs]: Can we use `from_delegate` here?
Receiving a `reject` message with `from_delegate`, the leader knows it comes from a delegate.
And as a delegated follower don't send `reject` message to the leader, the `send_append` could be always false I think?

Comment (hicqu) [reply src/raft.rs]: Ok. I will use `from_delegate` here. BTW do you think we need to rename `from_delegate` to `delegate`?

**Pull request merged**